lib/kafka_ex/protocol/offset.ex

```diff
@@ -7,6 +7,21 @@ defmodule KafkaEx.Protocol.Offset do
   defmodule Response do
     defstruct topic: "", partition_offsets: []
     @type t :: %Response{topic: binary, partition_offsets: list}
+
+    @doc """
+    Returns the numeric offset of the first partition in the response,
+    or nil if the topic is not found.
+
+    This is a convenience for the most typical case.
+    """
+    @spec first_partition_offset(atom | t) :: number | nil
+    def first_partition_offset(:topic_not_found) do
+      nil
+    end
+    def first_partition_offset([%Response{partition_offsets: partition_offsets}]) do
+      first_partition = hd(partition_offsets)
+      first_partition.offset |> hd
+    end
```

Comment:
I kind of agree with you on not exposing any functions from inside the protocol. But I think the same argument kind of applies to exposing data structures from inside the protocol.  This helper function was intended to at least hide the details of the underlying data structure.

I can move this function, it's not a huge deal.  But I think if you want to (rightly) encapsulate the details of the protocol, we should find a way to expose the data via an API.  Maybe publicly-facing data structures to encapsulate the salient details that shouldn't change.